sql/core/src/main/scala/org/apache/spark/sql/dynamicpruning/PlanDynamicPruningFilters.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.dynamicpruning
```

Comment:
we shouldn't create a top level package for non-user facing stuff. Either put this into catalyst or execution.

Reply:
maybe put it in execution.adaptive

Reply:
This isn't about PlanDynamicPruningFilter, which could have been put in sql.execution. It's more about `PartitionPruning`, which has to match logical scan nodes, and they are in spark-sql (which I don't like but can't do anything about). Then I was left to struggle where to put this logical rule in spark-sql. Package `execution` seems more for physical stuff, and definitely I wouldn't want to mix DPP classes in the `adaptive` package.

Reply:
OK in either case I wouldn't put it as a new top level package.

Reply:
I'll submit a follow-up to move the package.